### PR TITLE
Modernize build, Platform Parity

### PR DESCRIPTION
**Pull request by sgtcoolguy**
**Description**:
- Use SDK 8.2.1.GA to build (need to use an 8_2_X branch build for now)
- Add apidocs
- add unit tests
- Add danger.JS to PRs
- fixes to build with SDK 8+
- set up package.json
- validate apidocs
- Add more supported formats (and new constants to module/docs)
- Move iOS back to ZXing (https://github.com/zxingify/zxingify-objc) from AVFoundation
- Support passing along decoding hints
- Added lots of development/workflow tweaks (husky, lint-staged, clang-format linting/formatting of source, conventional commit enforcement)
- Bump major version and denote breaking changes in CHANGELOG.md

### CODE REVIEW

**Comment by m1ga** [comment]
@sgtcoolguy another option would be moving the Android part to https://firebase.google.com/docs/ml-kit/android/read-barcodes as a library. Not sure about the performance or other advantages. But it is supported by google so it might have a better support on phones

File: ios/Classes/TiBarcodeViewController.m [hunk 1]
```diff
@@ -43,25 +41,32 @@ - (instancetype)initWithObjectTypes:(NSArray *)objectTypes
...
     if (_showRectangle) {
       CGRect rect = _overlayView.cropRect;
-      __weak TiBarcodeViewController *weakSelf = self;
-      [_scanner setDidStartScanningBlock:^(void) {
-        [[weakSelf scanner] setScanRect:rect];
-      }];
+      //      __weak TiBarcodeViewController *weakSelf = self;
+      //      [_scanner setDidStartScanningBlock:^(void) {
+      //        [[weakSelf scanner] setScanRect:rect];
+      //      }];
```
Comment (vijaysingh-axway) [inline ios/Classes/TiBarcodeViewController.m]: Commented lines should be removed.

File: ios/Classes/TiBarcodeViewController.m [hunk 2]
```diff
@@ -89,26 +94,107 @@ - (UIInterfaceOrientation)preferredInterfaceOrientationForPresentation
...
+
   return YES;
 }
 
 - (void)handleDeviceRotation:(NSNotification *)notification
 {
   if (_showRectangle) {
     CGRect rect = _overlayView.cropRect;
-    [_scanner setScanRect:rect];
+    // [_scanner setScanRect:rect];
```
Comment (vijaysingh-axway) [inline ios/Classes/TiBarcodeViewController.m]: I think commented line should be removed (Or did you commented it to revisit ? ).

**Comment by garymathews** [comment]
Pushed changes to remove commented code. Tested the provided example on both Android and iOS. Barcodes scan fine.

Merging.

**Pull request merged**